be/src/olap/olap_server.cpp

```diff
@@ -334,6 +334,30 @@ void StorageEngine::_compaction_tasks_producer_callback() {
     int64_t interval = config::generate_compaction_tasks_min_interval_ms;
     do {
         if (!config::disable_auto_compaction) {
+            VLOG_CRITICAL << "compaction thread pool. num_threads: " << _compaction_thread_pool->num_threads()
+                      << ", num_threads_pending_start: " << _compaction_thread_pool->num_threads_pending_start()
+                      << ", num_active_threads: " << _compaction_thread_pool->num_active_threads()
+                      << ", max_threads: " << _compaction_thread_pool->max_threads()
+                      << ", min_threads: " << _compaction_thread_pool->min_threads()
+                      << ", num_total_queued_tasks: " << _compaction_thread_pool->num_total_queued_tasks();
+
+            if(_compaction_thread_pool->max_threads() != config::max_compaction_threads) {
+                Status status = _compaction_thread_pool->set_max_threads(config::max_compaction_threads);
+                if (status.ok()) {
```

Comment:
> what if it returns failure?

Max and min thread num for compaction thread pool are both determined by config `max_compaction_threads`. It means we turn down the config `max_compaction_threads` if it returns failure. when we call `set_max_threads()`, `_min_threads` hasn't been adjusted yet and` _max_threads` can not be less than `_min_threads`, so it would return failure. We don't need to worry. `set_min_threads()`will be executed and turn down `_min_threads` later in this round of compaction producer. `set_max_threads()`will be executed successfully next round and turn down ` _max_threads`.